test/unit/application.spec.js

```diff
@@ -1,74 +1,219 @@
 
-      this.app.start(this.fooOptions);
+    beforeEach(function() {
+      startStub = this.sinon.stub();
+      fooOptions = {foo: 'bar'};
+      app = new Application();
+
+      app.on('start', startStub);
     });
 
     it('should run the onStart callback', function() {
-      expect(this.startStub).to.have.been.called;
+      app.start(fooOptions);
+
+      expect(startStub).to.have.been.called;
     });
 
     it('should pass the startup option to the callback', function() {
-      expect(this.startStub).to.have.been.calledOnce.and.calledWith(this.app, this.fooOptions);
+      app.start(fooOptions);
+
+      expect(startStub).to.have.been.calledOnce.and.calledWith(app, fooOptions);
     });
 
     it('should have a cidPrefix', function() {
-      expect(this.app.cidPrefix).to.equal('mna');
+      app.start(fooOptions);
+
+      expect(app.cidPrefix).to.equal('mna');
     });
 
     it('should have a cid', function() {
-      expect(this.app.cid).to.exist;
+      app.start(fooOptions);
+
+      expect(app.cid).to.exist;
     });
   });
 
-  describe('when initializing with regions', function() {
+  describe('#_getRegion', function() {
+    let app;
+    let fooOptions;
+
     beforeEach(function() {
-      this.fooOptions = {
+      fooOptions = {
         region: '#fixtures'
       };
-      this.view = new Marionette.View({
-        template: _.template('ohai')
-      });
-
-      this.app = new Marionette.Application(this.fooOptions);
-      this.app.showView(this.view);
+      app = new Application(fooOptions);
     });
+
     it('should be able to define region selectors as strings', function() {
-      expect(this.app._region.$el).to.have.length(1);
+      expect(app.getRegion().$el).to.have.length(1);
     });
 
     it('should get the region selector with getRegion', function() {
-      expect(this.app.getRegion().$el).to.have.length(1);
+      expect(app.getRegion().$el).to.have.length(1);
     });
+  });
+
+  describe('#_showView', function() {
+    let app;
+    let view;
+    let fooOptions;
+    let showViewInRegionSpy;
+
+    beforeEach(function() {
+      fooOptions = {
+        region: '#fixtures'
+      };
+      view = new View({
+        template: _.template('ohai')
+      });
+      app = new Application(fooOptions);
 
-    it('can show a view in its region', function() {
-      expect(this.app._region.el.innerHTML).to.contain('ohai');
+      const appRegion = app.getRegion();
+
+      showViewInRegionSpy = this.sinon.spy(appRegion, 'show');
     });
 
-    it('can use the getView function', function() {
-      expect(this.app.getView()).to.deep.equal(this.view);
+    describe('when additional arguments was passed', function() {
+      let fooArgs;
+
+      beforeEach(function() {
+        fooArgs = ['foo', 'bar'];
+      });
+
+      it('should call show method in region with additional arguments', function() {
+        app.showView(view, fooArgs);
+
+        expect(showViewInRegionSpy).to.have.been.called;
```

Comment:
Totally agree, it isn't. I cheated here living just expecting that `region.show` will be just called. Sorry, I couldn't stub `region.show` so I had to leave some comment or smth. I'll try to make it works.